src/pages/index.js

```diff
@@ -92,56 +138,145 @@ function getCardElement(data) {
     previewModalImageEl.alt = data.name;
   });
 
-  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
   cardDeleteBtn.addEventListener("click", () => {
-    cardElement.remove();
+    selectedCard = cardElement;
+    selectedCardId = cardId;
+    openModal(deleteModal);
   });
 
   return cardElement;
 }
 
-// Handle profile edit form submission
-function handleEditFormSubmit(e) {
-  e.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+function handleDeleteSubmit() {
+  if (selectedCard && selectedCardId) {
+    const submitButton = deleteBtn;
+    submitButton.textContent = "Deleting...";
+    submitButton.disabled = true;
+
+    api.deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove();
+        closeModal(deleteModal);
+        disabledButton(deleteBtn, settings); // Disable after successful deletion
+      })
+      .catch((err) => {
+        console.error(err);
+        submitButton.disabled = false; // Re-enable on error
```

Comment:
`submitButton.disabled = false;` should be in `.finally`, to enable the confirmation button in any case